flang/lib/Lower/Bridge.cpp

```diff
@@ -1482,8 +1483,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       if (eval.isActionStmt()) {
         successor = eval.controlSuccessor;
       } else if (eval.isConstruct()) {
-        assert(!eval.evaluationList->empty() && "empty construct eval list");
-        if (eval.evaluationList->back()
+        if (eval.getLastNestedEvaluation()
```

Comment:
Since the assert is no longer needed, it would be better to flatten the condition -

`} else if (eval.isConstruct() && eval.getLastNestedEvaluation()) {`

Reply:
Good point, done.